cmd/pulumi-test-language/interface.go

```diff
@@ -104,7 +113,11 @@ type languageTestServer struct {
 	done   chan error
 	addr   string
 
-	sdkLock sync.Mutex
+	mutexMapLock sync.Mutex
+	sdkLock      map[string]*sync.Mutex
+
+	artifactMapLock sync.RWMutex
+	artifactMap     map[string]string
```

Comment:
`gsync.Map[K, V]` is a thing, saves you managing your own mutex

Reply:
I switched to `gsync.Map` now, which makes this pretty clean and understandable.  

Spawning a goroutine would not give us any of the performance benefits here, though we could still do the caching.  I don't think it's worth it, over `gsync.Map`, which is pretty simple and straightforward.